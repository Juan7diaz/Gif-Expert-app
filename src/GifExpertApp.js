import React, {useState} from 'react'
import AddCategory from './components/AddCategory'
import NavBar from './components/NavBar'
import GifGrid from './components/GifGrid'

const GifExpertApp = ({ defaultCategories = ['shingeki no kyojin'] }) => {

    const [categories, setCategories] = useState( defaultCategories )
    const [cantGifs, setCantGifs ] = useState(5);

    return (
        <div>

            <NavBar title={"GifExpertApp"}/>
            <AddCategory setCategories={ setCategories } setCantGifs={ setCantGifs } />
            <hr />

            <ol>
                {
                    categories.map( category => (
                        <GifGrid
                            key={category}
                            category={ category }
                            cantGifs = { cantGifs }
                        />
                    ))
                }
            </ol>

        </div>
    )
}

export default GifExpertApp
