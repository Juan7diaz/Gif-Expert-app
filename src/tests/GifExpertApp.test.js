import '@testing-library/jest-dom'
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';


describe('pruebas al componente principal', () => {
    

    test('debe de mostrar correctamente GifExpertApp ', () => {
        
        const wrapper = shallow( <GifExpertApp /> );
        expect( wrapper ).toMatchSnapshot();

    })
    
    test('debe de mostrar una lista de categorias ', () => {
        
        const categories = ['shingeki no kyojin', 'The promised neverland' ];
        const wrapper = shallow( <GifExpertApp defaultCategories={ categories } /> );
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( 2 );

    })

    

})
