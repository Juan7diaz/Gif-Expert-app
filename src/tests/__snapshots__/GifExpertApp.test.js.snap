// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas al componente principal debe de mostrar correctamente GifExpertApp  1`] = `
<div>
  <NavBar
    title="GifExpertApp"
  />
  <AddCategory
    setCantGifs={[Function]}
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      cantGifs={5}
      category="shingeki no kyojin"
      key="shingeki no kyojin"
    />
  </ol>
</div>
`;

exports[`pruebas al componente principal debe de mostrar una lista de categorias  1`] = `
<div>
  <NavBar
    title="GifExpertApp"
  />
  <AddCategory
    setCantGifs={[Function]}
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      cantGifs={5}
      category="shingeki no kyojin"
      key="shingeki no kyojin"
    />
    <GifGrid
      cantGifs={5}
      category="The promised neverland"
      key="The promised neverland"
    />
  </ol>
</div>
`;
